model_analyzer/monitor/dcgm/model.py

```diff
@@ -0,0 +1,158 @@
+
+        # Create DCGM monitor group
+        self.group_id = dcgm_agent.dcgmGroupCreate(dcgm_handle,
+                                                   structs.DCGM_GROUP_EMPTY,
+                                                   "triton-monitor")
+        # Add the GPUs to the group
+        for gpu in self._gpus:
+            dcgm_agent.dcgmGroupAddDevice(dcgm_handle, self.group_id,
+                                          gpu.device_id())
+
+        # Is the background thread active
+        self._thread_active = False
+
+        # Background thread collecting results
+        self._thread = None
+
+        # Thread pool
+        self._thread_pool = ThreadPool(processes=1)
+
+    def start_recording_metrics(self, tags):
+        """Start recording a list of tags
+
+        Parameters
+        ----------
+        tags : list
+            Name of the metrics to be collected. Options include
+            *memory* and *compute*.
+        """
+        dcgm_handle = self.dcgm_handle
+        group_id = self.group_id
+        frequency = int(self._frequency * 1000)
+
+        # Metrics to watch for
+        fields = []
+        if 'memory' in tags:
+            fields += [
+                dcgm_fields.DCGM_FI_DEV_FB_FREE,
+                dcgm_fields.DCGM_FI_DEV_FB_TOTAL,
+                dcgm_fields.DCGM_FI_DEV_FB_USED
+            ]
+
+        self.dcgm_field_group_id = dcgm_agent.dcgmFieldGroupCreate(
+            dcgm_handle, fields, 'triton-monitor')
+
+        self.group_watcher = dcgm_field_helpers.DcgmFieldGroupWatcher(
+            dcgm_handle, group_id, self.dcgm_field_group_id.value,
+            structs.DCGM_OPERATION_MODE_MANUAL, frequency, 3600, 0, 0)
+
+        self._thread = self._thread_pool.apply_async(self._monitoring_loop,
+                                                     (tags,))
+
+    def _monitoring_loop(self, tags):
+        self._thread_active = True
+        frequency = self._frequency
+
+        while self._thread_active:
+            begin = time.time()
+            self.group_watcher.GetMore()
+            duration = time.time() - begin
+            if duration < frequency:
+                time.sleep(frequency - duration)
+
+        record_collector = RecordCollector()
+        for gpu in self._gpus:
+            device_id = gpu.device_id()
+            metrics = self.group_watcher.values[device_id]
+            if 'memory' in tags:
+                num_metrics = len(
+                    metrics[dcgm_fields.DCGM_FI_DEV_FB_FREE].values)
+                for i in range(num_metrics):
+                    free_memory = metrics[
+                        dcgm_fields.DCGM_FI_DEV_FB_FREE].values[i]
+                    used_memory = metrics[
+                        dcgm_fields.DCGM_FI_DEV_FB_USED].values[i]
+                    total_memory = metrics[
+                        dcgm_fields.DCGM_FI_DEV_FB_TOTAL].values[i]
+                    memory_record = GPUMemoryRecord(gpu, used_memory.value,
```

Comment:
Merging [this PR](https://github.com/triton-inference-server/model_analyzer/pull/14) will break this. Not sure how cases like this are generally handled. Would you rebase on my branch, or have mine merged first then rebase? In any case this needs to be changed to fit the new format.

Reply:
I can wait until your PR is merged and then rebase.